src/client/Microsoft.Identity.Client/OAuth2/OAuth2Client.cs

```diff
@@ -89,6 +99,27 @@ internal async Task<T> ExecuteRequestAsync<T>(Uri endPoint, HttpMethod method, R
 
                 if (response.StatusCode != HttpStatusCode.OK || expectErrorsOn200OK)
                 {
+                    //Check if we can perform device auth
+                    if (_deviceAuthManager != null && IsDeviceAuthChallenge(response, respondToDeviceAuthChallenge))
```

Comment:
Please have a look at the `NULL object pattern` (e.g. [here](https://www.c-sharpcorner.com/article/null-object-design-pattern/)). 

Instead of littering the code with `if (_deviceManager != null) then ...` , we can implement a NullDeviceManager : IDeviceManager. This object's behavior is "do nothing". It will provide a much better abstraction.